src/main/java/cleo/task/Task.java

```diff
@@ -0,0 +1,70 @@
+package cleo.task;
+
+/**
+ * Represents an abstract task.
+ * A task contains a description and a status indicating whether it is done or not.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    /**
+     *  Constructs a new Task with the given description.
+     * The task is initially marked as not done.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon based on the value of the `isDone` flag.
+     *
+     * @return the status icon as a string, either "[X] " if `isDone` is true, or "[ ] " if `isDone` is false.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "[X] " : "[ ] ");
+    }
+
+    /**
+     * Returns the description of the task as a string.
+     *
+     * @return the description of the task.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Returns the value of the `isDone` flag.
+     *
+     * @return the value of the `isDone` flag.
+     */
+    public Boolean isDoneString() {
```

Comment:
I think there may be a typo here. You are returning the wrapper class, Boolean, instead of the base boolean data type

Reply:
Thank you for spotting this! I've changed it accordingly :)